launch.go

```diff
@@ -268,6 +268,11 @@ func createEnvironment(base map[string]string, secrets screwdriver.Secrets) []st
 	for k, v := range combined {
 		envStrings = append(envStrings, strings.Join([]string{k, v}, "="))
 	}
+
+	for k, v := range build.Environment {
```

Comment:
Hmm yeah, we should. But this PR doesn't change any previous behaviors, it just refactors.